vpr/src/route/router_lookahead_map.cpp

```diff
@@ -473,7 +473,7 @@ static void compute_router_wire_lookahead(const std::vector<t_segment_inf>& segm
             if (!sample_nodes[chan_type].empty()) continue;
 
             //Try an exhaustive search to find a suitable sample point
-            for (int inode = 0; inode < int(device_ctx.rr_nodes.size()); ++inode) {
+            for (int inode = 0; inode < int(device_ctx.rr_nodes.size()); ++inode) { // StrongId range is already used in this source file, so it can't be used here
```

Comment:
@ethanroj23 Would you mind elaborating this comments? If we use strong id, we can delete LINE 477. And we replace this line as ``for (const RRNodeId& rr_node : temp_rr_graph.nodes()) {``

I am asking so because we will deprecate the ``rr_nodes`` in ``device_ctx`` once the refactoring is done. If you can provide more detailes, I can help.

Reply:
@tangxifan I have been running into issues when I use `for (const RRNodeId& rr_node : temp_rr_graph.nodes()) {` here because `for (RREdgeId edge : rr_graph.edge_range(parent))` on line 659 already defines the `vtr::StrongIdRange template` I'm sure there is a workaround, but I am not overly familiar with templates, so I haven't found a solution yet.

Reply:
@ethanroj23 I suspect it could be due to the workaround in https://github.com/verilog-to-routing/vtr-verilog-to-routing/blob/6aa52095002bbb5f52d3e68574d52ab799e63a25/vpr/src/route/router_lookahead_map.cpp#L651-L652

In PR #1930, we will finish all the edge-related APIs and the workaround can be avoided. 
Also, the ``edge_range`` APIs is already in the master
Would you mind giving a try on reworking the LINE 659 by using the RRGraphView data structure? 
If I miss anything in yesterday's meeting, please feel free to let me know.